Firefox Selection Fix.sh

```diff
@@ -66,16 +81,43 @@ function fix_firefox(){
   echo
   cd /tmp || exit
   mkdir omni
+
+  ### Cries on macOS some warnings and errors:
+  # warning [/Applications/Firefox.app/Contents/Resources/browser/omni.ja]:
+  #     48078620 extra bytes at beginning or within zipfile (attempting to process anyway)
+  # error [/Applications/Firefox.app/Contents/Resources/browser/omni.ja]:
+  #     reported length of central directory is -48078620 bytes too long (Atari STZip zipfile?  J.H.Holm ZIPSPLIT 1.1 zipfile?).  Compensating...
   unzip -q "$firefox_dir/browser/omni.ja" -d omni
-  sed -i 's/this\._preventClickSelectsAll = this\.focused;/this._preventClickSelectsAll = true;/' omni/modules/UrlbarInput.jsm
-  sed -i 's/this\._preventClickSelectsAll = this\._textbox\.focused;/this._preventClickSelectsAll = true;/' omni/chrome/browser/content/browser/search/searchbar.js
-  sed -i 's/return shortkey\.split[(]"_"[)]\[1\];/return (shortkey || "_").split("_")[1];/' omni/chrome/devtools/modules/devtools/client/definitions.js # See https://github.com/SebastianSimon/firefox-selection-fix/issues/2
+
+  ### macOS High Sierra sed syntax:
+  # sed [-Ealn] [-e command] [-f command_file] [-i extension] [file ...]
+  # -i extension Edit files in-place, saving backups with the specified extension.
+  #    If a zero-length extension is given, no backup will be saved.
+  sed -i '' 's/this\._preventClickSelectsAll = this\.focused;/this._preventClickSelectsAll = true;/' omni/modules/UrlbarInput.jsm
+  sed -i '' 's/this\._preventClickSelectsAll = this\._textbox\.focused;/this._preventClickSelectsAll = true;/' omni/chrome/browser/content/browser/search/searchbar.js
+  sed -i '' 's/return shortkey\.split[(]"_"[)]\[1\];/return (shortkey || "_").split("_")[1];/' omni/chrome/devtools/modules/devtools/client/definitions.js # See https://github.com/SebastianSimon/firefox-selection-fix/issues/2
+
   cd omni || exit
+
+  ### Official method of creating *.ja files:
+  # https://developer.mozilla.org/en-US/docs/Mozilla/About_omni.ja_(formerly_omni.jar)
```

Comment:
This isn’t necessary to explain here. It’s already explained in the Super User answer.